### ISSUE TITLE
Wrong Timezone in Tasks?

### ISSUE THREAD

**Issue by vatolin** [nextcloud/tasks#2586]
### Steps to reproduce

1. Create a new task.
2. Set starting date at e.g. 12:00
3. Click at "due at".

At this click the time, given at step 2 jumps back by two hours. Obviously Tasks operate with UTC and does not recognize that the timezone of the server is CEST.

Is this a bug? Or is it a misconfiguration of the server? But I am sure that in Tasks before version 0.16.0 the given starting time did not jump back.

### Expected behavior

I would expect, that the given staring time would being kept by.

### Actual behaviour

The given starting time is being replaced by a time two hours back.

### Tasks app version

0.16.0

### CalDAV-clients used

Web Interface of NC

### Browser

Firefox 126.0.0

### Client operating system

macOS 12.7.5

### Server operating system

Linux 5.10.0-29-amd64 x86_64

### Web server

Apache

### Database engine version

PostgreSQL

### PHP engine version

PHP 8.2

### Nextcloud version

29.0.1

### Updated from an older installed version or fresh install

Updated from an older version

### List of activated apps

```Text
I can provide this information later, probably.
```

### Nextcloud configuration

```Text
I can provide this information later, probably.
```

### Web server error log

```Text
I do not really know, how to finde the appropriate log file.
```

### Log file

```Text
I do not really know, how to finde the appropriate log file.
```

### Browser log

```Text
I do not really know, how to finde the appropriate log file.
```

**Comment by philiprenich** [comment]
I'm seeing this as well. Time is changed from my input time +1200 (say, 6:30pm) to UTC (6:30am)

It looks like 0.16 was [supposed to have a fix for this](https://github.com/nextcloud/tasks/pull/2446), but maybe it's a different issue that causes this regression?

**Comment by ma-ivanov** [comment]
Confirm that this issue exists in my setup as well

**Comment by sidgames5** [comment]
I can confirm this issue

**Comment by jmarmstrong1207** [comment]
Can also confirm this, but changes to a different time offset from OP's. The correct time shows when I refresh the page.

**Comment by JD145** [comment]
I have the same wrong behavior since the last Tasks release.

**Comment by AlphaCraft9658** [comment]
I also experience similar things. When creating a task with a due-date and time, it displays in the wrong time zone, two hours before the actual intended time. But when I refresh the page, it's fixed. It's a minor bug, but still annoying. It looks like there have been similar things before, which are already fixed, but it's still here. Unless it has been fixed with Nextcloud 29 already, I can confirm this bug. (I couldn't update to NC 29 yet, as AIO hasn't pushed it yet.)

**Comment by philiprenich** [comment]
> (I couldn't update to NC 29 yet, as AIO hasn't pushed it yet.)

[See this discussion](https://github.com/nextcloud/all-in-one/discussions/4542) on updating to NC29 while on AIO, if you'd like to do it yourself before it lands in AIO.

**Comment by AlphaCraft9658** [comment]
> > (I couldn't update to NC 29 yet, as AIO hasn't pushed it yet.)
> 
> [See this discussion](https://github.com/nextcloud/all-in-one/discussions/4542) on updating to NC29 while on AIO, if you'd like to do it yourself before it lands in AIO.

I tried it before, but it caused issues. I am going to wait.

**Comment by applePrincess** [comment]
I feel like the expected behaviour is

1. You can configure the time zone in the settings found in bottom left corner of the app (so per user)
2. the default time zone is where the server is (or configured in the admin page. config.php even?)
3. use that info to display/update time.

I guess this is similar to the calendar app

**Comment by Emperor-Ovaltine** [comment]
I can confirm the same situation is happening on my setup as well.  Rebooting the system will cause it to display the correct time, (pulling from the system clock), however editing the task to set a start/end time will default back to the wrong time zone.

**Comment by syntacs** [comment]
System reboot is not necessary, browser reload is sufficient here.

**Comment by Emperor-Ovaltine** [comment]
> System reboot is not necessary, browser reload is sufficient here.

Further testing on my end confirms this on my setup. Refreshing the page in the browser also fixed the displayed time and the entire reboot was not needed.

**Comment by raimund-schluessler** [comment]
I packed a version which hopefully has this issue fixed: [tasks.tar.gz](https://github.com/user-attachments/files/16501909/tasks.tar.gz)

It would be very helpful if someone could test it to confirm the issue being fixed.

**Comment by syntacs** [comment]
not fixed here

- Nextcloud Hub 8 (29.0.4)

- nginx 1:1.24.0-1.fc37

- php-fpm 8.1.25-1.fc37

diff -r --brief tasks_orig tasks | while read line; do   file=$(echo $line | awk '{print $4}');   if [ -f "$file" ]; then     ls -l --time-style=long-iso "$file";   else     echo $line;   fi; done

-rwxr-xr-x. 1 apache apache 1971 2024-08-04 18:18 tasks/appinfo/info.xml
-rwxr-xr-x. 1 apache apache 37803 2024-08-05 21:22 tasks/appinfo/signature.json
-rwxr-xr-x. 1 apache apache 100451 2024-08-05 21:22 tasks/css/tasks-main.css
-rwxr-xr-x. 1 apache apache 5430 2024-08-04 18:18 tasks/img/favicon.ico
-rwxr-xr-x. 1 apache apache 2138 2024-08-04 18:18 tasks/img/favicon.png
-rwxr-xr-x. 1 apache apache 487 2024-08-04 18:18 tasks/img/favicon.svg
-rwxr-xr-x. 1 apache apache 3224 2024-08-04 18:18 tasks/img/favicon-touch.png
-rwxr-xr-x. 1 apache apache 515 2024-08-04 18:18 tasks/img/favicon-touch.svg
-rwxr-xr-x. 1 apache apache 356 2024-08-04 18:18 tasks/img/tasks-dark.svg
-rwxr-xr-x. 1 apache apache 1836 2024-08-04 18:18 tasks/img/tasks.png
-rwxr-xr-x. 1 apache apache 366 2024-08-04 18:18 tasks/img/tasks.svg
Only in tasks/js: Plus-BFnWochT.mjs
Only in tasks/js: Plus-BFnWochT.mjs.map
Only in tasks_orig/js: Plus--OvEy1Tw.mjs
Only in tasks_orig/js: Plus--OvEy1Tw.mjs.map
Only in tasks/js: store-BAR8kSTD.mjs
Only in tasks/js: store-BAR8kSTD.mjs.map
Only in tasks_orig/js: store-DE2z1cMA.mjs
Only in tasks_orig/js: store-DE2z1cMA.mjs.map
Only in tasks_orig/js: TaskCreateDialog-D9vIR_Ml.mjs
Only in tasks_orig/js: TaskCreateDialog-D9vIR_Ml.mjs.map
Only in tasks/js: TaskCreateDialog-Db4kU-tc.mjs
Only in tasks/js: TaskCreateDialog-Db4kU-tc.mjs.map
-rwxr-xr-x. 1 apache apache 9835 2024-08-05 21:22 tasks/js/tasks-dashboard.mjs
-rwxr-xr-x. 1 apache apache 33435 2024-08-05 21:22 tasks/js/tasks-dashboard.mjs.map
-rwxr-xr-x. 1 apache apache 647191 2024-08-05 21:22 tasks/js/tasks-main.mjs
-rwxr-xr-x. 1 apache apache 2182916 2024-08-05 21:22 tasks/js/tasks-main.mjs.map
-rwxr-xr-x. 1 apache apache 1199 2024-08-05 21:22 tasks/js/tasks-talk.mjs
-rwxr-xr-x. 1 apache apache 6298 2024-08-05 21:22 tasks/js/tasks-talk.mjs.map
-rwxr-xr-x. 1 apache apache 3222 2024-08-05 21:22 tasks/js/vendor.LICENSE.txt
-rwxr-xr-x. 1 apache apache 23734 2024-08-04 18:18 tasks/l10n/ar.js
-rwxr-xr-x. 1 apache apache 23729 2024-08-04 18:18 tasks/l10n/ar.json
-rwxr-xr-x. 1 apache apache 1826 2024-08-04 18:18 tasks/l10n/br.js
-rwxr-xr-x. 1 apache apache 1821 2024-08-04 18:18 tasks/l10n/br.json
-rwxr-xr-x. 1 apache apache 18599 2024-08-04 18:18 tasks/l10n/de_DE.js
-rwxr-xr-x. 1 apache apache 18594 2024-08-04 18:18 tasks/l10n/de_DE.json
-rwxr-xr-x. 1 apache apache 16710 2024-08-04 18:18 tasks/l10n/en_GB.js
-rwxr-xr-x. 1 apache apache 16705 2024-08-04 18:18 tasks/l10n/en_GB.json
-rwxr-xr-x. 1 apache apache 2150 2024-08-04 16:43 tasks/l10n/es_CL.js
-rwxr-xr-x. 1 apache apache 2145 2024-08-04 16:43 tasks/l10n/es_CL.json
-rwxr-xr-x. 1 apache apache 18659 2024-08-04 16:43 tasks/l10n/es.js
-rwxr-xr-x. 1 apache apache 18654 2024-08-04 16:43 tasks/l10n/es.json
-rwxr-xr-x. 1 apache apache 4364 2024-08-04 18:18 tasks/l10n/es_MX.js
-rwxr-xr-x. 1 apache apache 4359 2024-08-04 18:18 tasks/l10n/es_MX.json
-rwxr-xr-x. 1 apache apache 18127 2024-08-04 18:18 tasks/l10n/eu.js
-rwxr-xr-x. 1 apache apache 18122 2024-08-04 18:18 tasks/l10n/eu.json
Only in tasks/l10n: ga.js
Only in tasks/l10n: ga.json
-rwxr-xr-x. 1 apache apache 18123 2024-08-04 18:18 tasks/l10n/gl.js
-rwxr-xr-x. 1 apache apache 18118 2024-08-04 18:18 tasks/l10n/gl.json
-rwxr-xr-x. 1 apache apache 13350 2024-08-04 16:43 tasks/l10n/he.js
-rwxr-xr-x. 1 apache apache 13345 2024-08-04 16:43 tasks/l10n/he.json
-rwxr-xr-x. 1 apache apache 6610 2024-08-04 18:18 tasks/l10n/ko.js
-rwxr-xr-x. 1 apache apache 6605 2024-08-04 18:18 tasks/l10n/ko.json
-rwxr-xr-x. 1 apache apache 3713 2024-08-04 16:43 tasks/l10n/lv.js
-rwxr-xr-x. 1 apache apache 3708 2024-08-04 16:43 tasks/l10n/lv.json
-rwxr-xr-x. 1 apache apache 17447 2024-08-04 18:18 tasks/l10n/nb.js
-rwxr-xr-x. 1 apache apache 17442 2024-08-04 18:18 tasks/l10n/nb.json
-rwxr-xr-x. 1 apache apache 18735 2024-08-04 16:43 tasks/l10n/pl.js
-rwxr-xr-x. 1 apache apache 18730 2024-08-04 16:43 tasks/l10n/pl.json
-rwxr-xr-x. 1 apache apache 18640 2024-08-04 18:18 tasks/l10n/pt_BR.js
-rwxr-xr-x. 1 apache apache 18635 2024-08-04 18:18 tasks/l10n/pt_BR.json
-rwxr-xr-x. 1 apache apache 23495 2024-08-04 18:18 tasks/l10n/sr.js
-rwxr-xr-x. 1 apache apache 23490 2024-08-04 18:18 tasks/l10n/sr.json
-rwxr-xr-x. 1 apache apache 10625 2024-08-04 16:43 tasks/l10n/sv.js
-rwxr-xr-x. 1 apache apache 10620 2024-08-04 16:43 tasks/l10n/sv.json
-rwxr-xr-x. 1 apache apache 15257 2024-08-04 16:43 tasks/l10n/zh_CN.js
-rwxr-xr-x. 1 apache apache 15252 2024-08-04 16:43 tasks/l10n/zh_CN.json
-rwxr-xr-x. 1 apache apache 16327 2024-08-04 18:18 tasks/l10n/zh_HK.js
-rwxr-xr-x. 1 apache apache 16322 2024-08-04 18:18 tasks/l10n/zh_HK.json
-rwxr-xr-x. 1 apache apache 16688 2024-08-04 18:18 tasks/l10n/zh_TW.js
-rwxr-xr-x. 1 apache apache 16683 2024-08-04 18:18 tasks/l10n/zh_TW.json

**Comment by raimund-schluessler** [comment]
> not fixed here

Are you sure that not an old version of Tasks is cashed somewhere (either in your browser, or on the server)? It works fine here with 29.0.4 and the version from above.

**Comment by syntacs** [comment]
I used the download in ur post
Can you check the file size of two or three files which are important for the fix ?  That's why I included the complete 'ls'

**Comment by syntacs** [comment]
Update: all fine after deleting browser data manually. It is working now

**Comment by philiprenich** [comment]
NC 29.0.4 with Tasks 0.16 and I'm still seeing the incorrect timezone initially. Set completed or start date to 5:00pm and it swaps to 5:00am (I'm +1200 UTC, so this is the "expected incorrect" time). A refresh displays the time correctly. So the server is saving it fine, it's just things get confused when the UI handles the input directly.

I've dumped browser data as well as kept dev tools open with "don't load from cache".

There seem to be a few places where timezones are handled, so I'm wondering if this is one last place that needs fixing?

**Comment by raimund-schluessler** [comment]
This issue got reported for 0.16.0 and the milestone is set to 0.16.1. This means it is not working for 0.16.0 and will be fixed in the next minor release 0.16.1.

**Comment by philiprenich** [comment]
> This issue got reported for 0.16.0 and the milestone is set to 0.16.1. This means it is not working for 0.16.0 and will be fixed in the next minor release 0.16.1.

Ah, my bad, misunderstood the previous messages. Thanks!
